### PR TITLE
Help: Handle external links

### LINKED ISSUE #2727: Help: Handle links to external websites
Apparently the Help application can't handle links to external websites. I would expect that a click on said link opens the website in the Browser - instead, a blank page is shown in the Help application.

cc @bugaevc

```
Help(23): Web::EventHandler: Clicking on a link to https://jestjs.io/
[Help(23:23)]: Rejecting path '/' since it hasn't been unveiled.
[Help(23:23)]: 0xc0143eef  Kernel::VFS::validate_path_against_process_veil(AK::StringView, int) +277
[Help(23:23)]: 0xc01443db  Kernel::VFS::resolve_path(AK::StringView, Kernel::Custody&, AK::RefPtr<Kernel::Custody>*, int, int) +163
[Help(23:23)]: 0xc016ac77  Kernel::Process::sys$realpath(Kernel::Syscall::SC_realpath_params const*) +401
[Help(23:23)]: 0xc0177da1  syscall_handler +1271
[Help(23:23)]: 0xc017784b  syscall_asm_entry +49
[Help(23:23)]: 0x0804c148  AK::Function<void (AK::URL const&, AK::String const&, unsigned int)>::CallableWrapper<main::{lambda(auto:1&, auto:2&, unsigned int)#5}>::call(AK::URL const&, AK::String const&, unsigned int) const +152
[Help(23:23)]: 0x080b5cd6  Web::EventHandler::handle_mousedown(Gfx::IntPoint const&, unsigned int, unsigned int) +1288
[Help(23:23)]: 0x0804ec87  Web::PageView::mousedown_event(GUI::MouseEvent&) +83
[Help(23:23)]: 0x080e6c8b  GUI::Widget::handle_mousedown_event(GUI::MouseEvent&) +49
[Help(23:23)]: 0x081717b2  Core::Object::dispatch_event(Core::Event&, Core::Object*) +74
[Help(23:23)]: 0x080ec670  GUI::Window::event(Core::Event&) +1070
[Help(23:23)]: 0x081717b2  Core::Object::dispatch_event(Core::Event&, Core::Object*) +74
[Help(23:23)]: 0x0816b9e9  Core::EventLoop::pump(Core::EventLoop::WaitMode) +307
[Help(23:23)]: 0x0816bb67  Core::EventLoop::exec() +43
[Help(23:23)]: 0x08048dbb  main +2515
[Help(23:23)]: 0x0804b53e  _start +94
```

### PR DESCRIPTION
**Pull request by linusg**
Links which do not have the "file" protocol (most commonly links to external websites - http and https) are now not treated as file paths anymore but instead handled by Desktop::Launcher.

Fixes #2727.

cc @bugaevc :)

### CODE REVIEW

**Comment by bugaevc** [comment]
I'd also open file URLs outside of `/usr/share/man` as external files (so, in Browser) too.

**Comment by linusg** [comment]
> I'd also open file URLs outside of /usr/share/man as external files (so, in Browser) too.

Great idea, done!
I also added a link to a relevant JS file in the `test-js(1)` man page which now opens in Text Editor. very cool :smile:

File: Applications/Help/main.cpp [hunk 1]
```diff
@@ -159,10 +165,21 @@ int main(int argc, char* argv[])
         model->update_section_node_on_toggle(index, open);
     };
 
+    auto open_external = [](auto& url) {
+        if (!Desktop::Launcher::open(url))
+            dbg() << "Failed to open '" << url << "'";
```
Comment (bugaevc) [inline Applications/Help/main.cpp:170]: We need to actually notify the user, not just log it. Perhaps show a message box?

File: Applications/Help/main.cpp [hunk 2]
```diff
@@ -159,10 +165,21 @@ int main(int argc, char* argv[])
...
+    };
+
     page_view.on_link_click = [&](auto& url, auto&, unsigned) {
-        char* current_path = strdup(history.current().characters());
-        char* path = realpath(url.path().characters(), nullptr);
```
Comment (bugaevc) [inline Applications/Help/main.cpp:164]: Yay for getting rid of the `realpath()` call! Hopefully one day we'll be able to make `Core::File::real_path_for()` not rely on `realpath()` (and `PATH_MAX`) internally :smile:

File: Base/usr/share/man/man1/test-js.md [hunk 3]
```diff
@@ -12,7 +12,8 @@ $ test-js [options...]
 
 `test-js` runs the LibJS test suite located in `/home/anon/js-tests`. These
 tests are using a custom JavaScript testing framework inspired by
-[Jest](https://jestjs.io) (see `test-common.js` in the same directory).
+[Jest](https://jestjs.io) (see [`test-common.js`](/home/anon/js-tests/test-common.js)
+in the same directory).
```
Comment (bugaevc) [inline Base/usr/share/man/man1/test-js.md:16]: Perhaps drop "in the same directory" then?

File: Applications/Help/main.cpp (hunk 1 above)
Comment (linusg) [reply Applications/Help/main.cpp]: I left in the added message box nonetheless.

**Pull request merged**